src/Core/Scheduler/Cache.js

```diff
@@ -134,18 +137,28 @@ class Cache {
      * @param {string} [key3]
      */
     delete(key1, key2, key3) {
-        const entry = this.data.get(key1);
-        if (entry == undefined) {
-            throw Error('Please specify at least a key of something to delete');
-        } else if (entry.get(key2) == undefined) {
+        const entry_1 = this.data.get(key1);
+        if (entry_1 !== undefined && !(entry_1 instanceof Map)) {
             delete this.data.get(key1);
             this.data.delete(key1);
-        } else if (entry.get(key2).get(key3) == undefined) {
-            delete entry.get(key2);
-            entry.delete(key2);
         } else {
-            delete entry.get(key2).get(key3);
-            entry.get(key2).delete(key3);
+            const entry_2 = entry_1.get(key2);
```

Comment:
This will break if `entry_1` is `undefined`.

Reply:
fixed